File: src/test/java/no/novari/flyt/history/repository/utils/performance/EventSequenceGenerationConfig.java
package no.novari.flyt.history.repository.utils.performance;

import lombok.Getter;
import lombok.ToString;
import no.novari.flyt.history.model.event.EventCategory;

import java.util.List;

@ToString
@Getter
public class EventSequenceGenerationConfig {
    private final List<EventCategory> eventSequence;
    private final int numberOfSequences;
    private final String sourceApplicationInstanceIdOverride;

    public EventSequenceGenerationConfig(
            List<EventCategory> eventSequence,
            int numberOfSequences,
            String sourceApplicationInstanceIdOverride
    ) {
        this.eventSequence = eventSequence;
        this.numberOfSequences = numberOfSequences;
        this.sourceApplicationInstanceIdOverride = sourceApplicationInstanceIdOverride;
    }

    public EventSequenceGenerationConfig(
            EventSequence eventSequence,
            int numberOfSequences,
            String sourceApplicationInstanceIdOverride
    ) {
        this(eventSequence.getOrder(), numberOfSequences, sourceApplicationInstanceIdOverride);
    }

    public EventSequenceGenerationConfig(
            List<EventCategory> eventSequence,
            int numberOfSequences
    ) {
        this(eventSequence, numberOfSequences, null);
    }

    public EventSequenceGenerationConfig(
            EventSequence eventSequence,
            int numberOfSequences
    ) {
        this(eventSequence.getOrder(), numberOfSequences);
    }

}

File: src/test/kotlin/no/novari/flyt/history/repository/utils/performance/EventSequenceGenerationConfig.kt
package no.novari.flyt.history.repository.utils.performance

import no.novari.flyt.history.model.event.EventCategory

class EventSequenceGenerationConfig(
    val eventSequence: List<EventCategory>,
    val numberOfSequences: Int,
    val sourceApplicationInstanceIdOverride: String? = null,
) {
    constructor(
        eventSequence: EventSequence,
        numberOfSequences: Int,
        sourceApplicationInstanceIdOverride: String?,
    ) : this(eventSequence.order, numberOfSequences, sourceApplicationInstanceIdOverride)

    constructor(
        eventSequence: List<EventCategory>,
        numberOfSequences: Int,
    ) : this(eventSequence, numberOfSequences, null)

    constructor(
        eventSequence: EventSequence,
        numberOfSequences: Int,
    ) : this(eventSequence.order, numberOfSequences)

    override fun toString(): String {
        return "EventSequenceGenerationConfig(eventSequence=$eventSequence, numberOfSequences=$numberOfSequences, " +
            "sourceApplicationInstanceIdOverride=$sourceApplicationInstanceIdOverride)"
    }
}
